x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -629,12 +668,49 @@ public boolean isAllowedByLicense(OperationMode minimumMode, boolean needActive)
 
     /**
      * A convenient method to test whether a feature is by license status.
-     * @see #isAllowedByLicense(OperationMode, boolean)
      *
-     * @param minimumMode  The minimum license to meet or exceed
+     * @param minimumMode The minimum license to meet or exceed
+     * @see #isAllowedByLicense(OperationMode, boolean)
      */
     public boolean isAllowedByLicense(OperationMode minimumMode) {
         return isAllowedByLicense(minimumMode, true);
     }
 
+    public static class FeatureUsage {
+        public final Feature feature;
+        public final long lastUsed;
+        public final Set<String> identifiers;
+
+        public FeatureUsage(Feature feature, long lastUsed, Set<String> identifiers) {
+            this.feature = feature;
+            this.lastUsed = lastUsed;
+            this.identifiers = Objects.requireNonNull(identifiers);
```

Comment:
IIUC, most feature usages have no identifier. Would it be better to allow `null` for `identifiers`, which in turn can be skipped in the final response returned to users instead of cluttering the response with `"identifiers": []`?